src/raftstore/store/worker/region.rs

```diff
@@ -206,8 +207,12 @@ impl SnapContext {
             write_batch_size: self.batch_size,
         };
         try!(s.apply(options));
+
+        let wb = WriteBatch::new();
         region_state.set_state(PeerState::Normal);
-        box_try!(self.db.put_msg(&region_key, &region_state));
+        box_try!(wb.put_msg(&region_key, &region_state));
+        box_try!(wb.delete(&keys::raft_state_key(region_id)));
```

Comment:
Both KV and Raft use raft_state_key are strange. 
Although the value is RaftLocalState, I don't think the key name should be raft state key. It is more proper with the name like raft_applying_state_key because this key can only exist when applying snapshot.